modulus/datapipes/climate/climate_hdf5.py

```diff
@@ -0,0 +1,590 @@
+                of fields {f['fields'].shape[1]}"
+                )
+
+            if self.num_samples_per_year > data_samples_per_year:
+                raise ValueError(
+                    f"num_samples_per_year ({self.num_samples_per_year}) > number of \
+                    samples available ({data_samples_per_year})!"
+                )
+
+            self.logger.info(f"Number of samples/year: {self.num_samples_per_year}")
+            self.logger.info(f"Number of channels available: {f['fields'].shape[1]}")
+
+    def _load_statistics(self) -> None:
+        """Loads climate statistics from pre-computed numpy files
+
+        The statistic files should be of name global_means.npy and global_std.npy with
+        a shape of [1, C, 1, 1] located in the stat_dir.
+
+        Raises
+        ------
+        IOError
+            If mean or std numpy files are not found
+        AssertionError
+            If loaded numpy arrays are not of correct size
+        """
+        # If no stats dir we just skip loading the stats
+        if self.stats_dir is None:
+            self.mu = None
+            self.std = None
+            return
+        # load normalisation values
+        mean_stat_file = self.stats_dir / Path("global_means.npy")
+        std_stat_file = self.stats_dir / Path("global_stds.npy")
+
+        if not mean_stat_file.exists():
+            raise IOError(f"Mean statistics file {mean_stat_file} not found")
+        if not std_stat_file.exists():
+            raise IOError(f"Std statistics file {std_stat_file} not found")
+
+        # has shape [1, C, 1, 1]
+        self.mu = np.load(str(mean_stat_file))[:, self.channels]
+        # has shape [1, C, 1, 1]
+        self.sd = np.load(str(std_stat_file))[:, self.channels]
+
+        if not self.mu.shape == self.sd.shape == (1, len(self.channels), 1, 1):
+            raise AssertionError("Error, normalisation arrays have wrong shape")
+
+    def _load_land_sea_mask(self) -> None:
+        """Load land-sea mask from netCDF file."""
+        ds = nc.Dataset(self.lsm_filename)
+        lsm = np.array(ds["lsm"]).astype(np.float32)
+        lsm = np.flip(
+            lsm, axis=1
+        )  # flip latitude axis, TODO hacky fix and we should get this from the file
+        assert (
+            lsm.shape[1:] == self.data_shape
+        ), "Land-sea mask shape does not match data shape"
+        lsm = lsm[:, : self.cropped_data_shape[0], : self.cropped_data_shape[1]]
+        self.lsm = dali.types.Constant(lsm)
+
+    def _load_geopotential(self, normalize: bool = True) -> None:
+        """Get geopotential from netCDF file."""
+        ds = nc.Dataset(self.geopotential_filename)
+        geop = np.array(ds["z"]).astype(np.float32)
+        geop = np.flip(
+            geop, axis=1
+        )  # flip latitude axis, TODO hacky fix and we should get this from the file
+        assert (
+            geop.shape[1:] == self.data_shape
+        ), "Geopotential shape does not match data shape"
```

Comment:
Again, a more informative error message can be
`f"Geopotential shape {geop.shape[1:]} does not match data shape {self.data_shape}"`

Reply:
Fixed this.